go/vt/topo/memorytopo/election.go

```diff
@@ -153,8 +153,12 @@ func (mp *cLeaderParticipation) WaitForNewLeader(ctx context.Context) (<-chan st
 	}
 
 	notifications := make(chan string, 8)
+
+	nextWatchIndexMu.Lock()
 	watchIndex := nextWatchIndex
 	nextWatchIndex++
+	nextWatchIndexMu.Unlock()
+
 	n.watches[watchIndex] = watch{lock: notifications}
```

Comment:
Hmm, not sure if an atomic increment is actually enough, but don't we have to add updating `n.watches` to the locked area as well? Because we'd then still write to a map from multiple goroutines in parallel?